cmake/HalideSanitizerHelpers.cmake

```diff
@@ -2,23 +2,29 @@
 # Utilities for setting up Sanitizer settings
 ##
 
-message(STATUS "TODO: DEFAULTING TO ON FOR NOW FOR DRAFT PR, PLEASE IGNORE")
-option(Halide_ENABLE_ASAN "Build everything with Address Sanitizer enabled." ON)
-if (Halide_ENABLE_ASAN)
-    message(STATUS "Enabling Address Sanitizer for project ${PROJECT_NAME}.")
-    set(SANITIZER_COMPILE_OPTIONS -fsanitize=address)
-    set(SANITIZER_LINK_OPTIONS -fsanitize=address $<$<CXX_COMPILER_ID:GNU>:-static-libasan>)
-    # If running under ASAN, we need to suppress some errors:
-    # - detect_leaks, because circular Expr chains in Halide can indeed leak,
-    #   but we don't care here
-    # - detect_container_overflow, because this is a known false-positive
-    #   if compiling with a non-ASAN build of LLVM (which is usually the case)
-    set(SANITIZER_ENV_VARS "ASAN_OPTIONS=detect_leaks=0:detect_container_overflow=0")
-    # -fPIC required for some ASAN build configurations (generally .so), so just enable it unversally here
-    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+if (Halide_CLANG_TIDY_BUILD)
+    message(STATUS "Ignoring all sanitizers because Clang-Tidy is in use.")
 else()
```

Comment:
Here and elsewhere: can't you just `return()`?
(In general, i find halide's style of else after-return to be confusing..)